src/utils/index.test.tsx

```diff
@@ -0,0 +1,279 @@
+
+    // This allows pending promises to resolve
+    await Promise.resolve()
+    expect(callback).toBeCalled()
+  })
+})
+
+describe('range', () => {
+  it('should return an array of numbers from start to end', () => {
+    const start = 1
+    const end = 5
+    const expected = [1, 2, 3, 4, 5]
+    expect(range(start, end)).toEqual(expected)
+  })
+
+  it('should return an empty array if start is greater than end', () => {
+    const start = 5
+    const end = 1
+    expect(range(start, end)).toEqual([])
+  })
+
+  it('should return an array with a single element if start equals end', () => {
+    const start = 5
+    const end = 5
+    expect(range(start, end)).toEqual([5])
+  })
+})
+
+describe('formatTime', () => {
+  it('should throw an error if seconds is a string or undefined', () => {
+    expect(() => formatTime('100')).toThrow('Should not call formatTime on a string')
+    expect(() => formatTime(undefined)).toThrow('Should not call formatTime on a string')
+  })
+
+  it('should return a formatted time string if seconds is a number', () => {
+    expect(formatTime(90)).toBe('01:30')
+    expect(formatTime(600)).toBe('10:00')
+  })
+})
+
+describe('breakpoints', () => {
+  it('should have correct values for sm, md, and lg', () => {
+    expect(breakpoints).toEqual({
+      sm: 650,
+      md: 800,
+      lg: 1100,
+    })
+  })
+})
+
+describe('Deferred', () => {
+  it('should resolve from input', async () => {
+    const input = 'test'
+    const deferred = new Deferred<string>()
+    deferred.resolve(input)
+    await expect(deferred.promise).resolves.toBe(input)
+  })
+
+  it('should reject from error', async () => {
+    const errorInput = 'test error'
+    const deferred = new Deferred<string>()
+    const error = new Error(errorInput)
+    deferred.reject(error)
+    await expect(deferred.promise).rejects.toThrow(errorInput)
+  })
+})
+
+describe('formatInstrumentName', () => {
+  it('should return an empty string if instrument is not passed', () => {
+    expect(formatInstrumentName(undefined as any)).toBe('')
+    expect(formatInstrumentName(null as any)).toBe('')
+    expect(formatInstrumentName('' as any)).toBe('')
+  })
+
+  it.each([
+    ['acoustic_grand_piano', 'Acoustic Grand Piano'],
+    ['acoustic_guitar_steel', 'Acoustic Guitar Steel'],
+    ['fx_2_soundtrack', 'Fx 2 Soundtrack'],
+  ])('should format instrument %s as title case', (instrumentSnakeCase, instrumentTitleCase) => {
+    expect(formatInstrumentName(instrumentSnakeCase as InstrumentName)).toBe(instrumentTitleCase)
+  })
+})
+
+describe('convertHexColorToIntArr', () => {
+  it('should return an empty array for input greather than 7 chars', () => {
+    const inputHex = '#fffffff'
+    expect(inputHex).toHaveLength(8)
+    expect(convertHexColorToIntArr(inputHex)).toEqual([])
+  })
+
+  it('should return an empty array for input lower than 7 chars', () => {
+    const inputHex = '#ffff'
+    expect(inputHex).toHaveLength(5)
```

Comment:
This line isn't testing anything
```suggestion
    expect(inputHex).toHaveLength(5)
```